crates/driver/src/logic/competition/auction/order.rs

```diff
@@ -0,0 +1,82 @@
+use {
+    crate::{
+        logic::{competition, eth},
+        util,
+    },
+    primitive_types::U256,
+};
+
+/// An order in the auction.
+#[derive(Debug)]
+pub struct Order {
+    pub uid: competition::OrderUid,
+    pub from: eth::Address,
+    pub receiver: Option<eth::Address>,
+    pub valid_to: util::Timestamp,
+    pub sell: eth::Asset,
+    pub buy: eth::Asset,
+    pub side: competition::Side,
+    pub fee: Fee,
+    pub kind: Kind,
+    pub app_data: competition::AppData,
+    /// A partial order doesn't require the full amount to be traded. E.g. only
+    /// 10% of the requested amount may be traded, if this leads to the most
+    /// optimal solution.
+    pub partial: bool,
+    /// The autopilot marks orders as mature after a certain time period. The
+    /// solvers can use heuristics on this field to optimize solution sizes.
+    pub mature: bool,
+    pub executed: U256,
```

Comment:
Would be nice to document that the executed amount refers to `sell` for sell orders and `buy` for buy orders.